src/GenFacades/GenFacades.Core/GenFacades.cs

```diff
@@ -433,7 +453,27 @@ public Assembly GenerateFacade(IAssembly contractAssembly, IAssemblyReference se
                         continue;
                     }
 
+                    if (buildPartialReferenceFacade)
+                    {
+                        // honor preferSeedType for keeping contract type
+                        string preferredSeedAssembly;
+                        bool keepType = _seedTypePreferences.TryGetValue(docId, out preferredSeedAssembly) &&
+                            contractAssemblyName.Equals(preferredSeedAssembly, StringComparison.OrdinalIgnoreCase);
+
+                        if (!keepType)
```

Comment:
If I'm reading this right, should we have an `else { continue; }` here? If we're keeping the existing type, then we don't need a type-forward, right? It might end up as a no-op later on, I guess.